### PR TITLE
148 fix target sda outputs

### PR DESCRIPTION
**Pull request by jdhoffa**
The original `target_sda` function had several places in which the output was unexpected. This was a result of basing the function off of a methodology document with errors. 

In this PR, I have already:
- [x] updated the `vignettes/target_sda.Rmd` to reflect the correct methodology (and reworded the article to make it more clear)
- [x] temporarily removed regression tests that reflect potentially incorrect output
- [x] added on LARGE regression test, that reflects the correct output (including a new possibility to output several targets for several scenarios)

- [x] work on `target_sda()` until that large test passes
- [ ] refactor that test into smaller more meaningful tests

Relates to #144, #145, #148

### CODE REVIEW

File: tests/testthat/test-target_sda.R [hunk 1]
```diff
@@ -228,3 +177,113 @@ test_that("outputs expected names", {
...
+
+expect_equal(out$projected$emission_factor_value, c(0.9, 0.9, 0.8, 0.5))
+expect_equal(out$corporate_economy$emission_factor_value, c(06.45, 0.9, 0.8, 0.5))
+expect_equal(out$adjusted_scenario_b2ds$emission_factor_value, c(6.45, 1.29))
+expect_equal(out$adjusted_scenario_sds$emission_factor_value, c(6.45, 5.16))
+expect_equal(out$target_b2ds$emission_factor_value, c(0.9, 1.29))
+expect_equal(out$target_sds$emission_factor_value, c(0.9, 5.16))
+
+
+})
```
Comment (jdhoffa) [inline tests/testthat/test-target_sda.R:237]: This is the test I am aiming to have pass

**Comment by jdhoffa** [comment]
@maurolepore there are a few TODO's, that I wanted your input on, and still that `tibble` R CMD CHECK warning, but other than that, this is more or less ready to be reviewed.

File: tests/testthat/test-target_sda.R [hunk 2]
```diff
@@ -208,23 +170,92 @@ test_that("outputs 3 metrics of CO2 emissions", {
...
+)
+
+out <- target_sda(matched,
+                  ald,
+                  co2_intensity_scenario) %>%
+  arrange(.data$year)
+
+out <- split(out, out$emission_factor_metric)
+
+expected_out <- tibble::tribble(
```
Comment (maurolepore) [inline tests/testthat/test-target_sda.R:204]: This is why R CMD check complained. We import `dplyr::tibble()`, not `tibble::tibble()`.

File: tests/testthat/test-target_sda.R [hunk 3]
```diff
@@ -208,23 +170,91 @@ test_that("outputs 3 metrics of CO2 emissions", {
...
+  "cement", 2025L,             "target_sds",                   5.16,
+  "cement", 2020L, "adjusted_scenario_b2ds",                   6.45,
+  "cement", 2025L, "adjusted_scenario_b2ds",                   1.29,
+  "cement", 2020L,  "adjusted_scenario_sds",                   6.45,
+  "cement", 2025L,  "adjusted_scenario_sds",                   5.16,
+  "cement", 2020L,      "corporate_economy",                   6.45,
+  "cement", 2021L,      "corporate_economy",                    0.9,
+  "cement", 2022L,      "corporate_economy",                    0.8,
+  "cement", 2025L,      "corporate_economy",                    0.5
+  )
```
Comment (maurolepore) [inline tests/testthat/test-target_sda.R:222]: Is this dead code? `expected_out` seems unused.

Comment (jdhoffa) [reply tests/testthat/test-target_sda.R]: Yup you are correct! Thanks

**Comment by jdhoffa** [comment]
Just had a look, and all the changes look good to me. Once we sort the regression value issues out after the team meeting, lets merge this :)

**Pull request merged**